### PR TITLE
Support passing the managed folder as an id or a Folder in setup_mlflow

### PR DESCRIPTION
**Pull request by alarive**
Also supports dataiku.Folder, as I saw no reason why we were not doing it.

Companion PR: https://github.com/dataiku/dip/pull/16126

### CODE REVIEW

**Comment by alarive** [comment]
Dip on demand failed because of timeout issues, looks unrelated to this PR, but I have restarted anyway:
http://dku13.dataiku.com:9180/job/dip-on-demand/8501

File: dataikuapi/dss_plugin_mlflow/utils.py [hunk 1]
```diff
@@ -62,30 +62,39 @@ def __init__(self, client, project, managed_folder, host=None):
...
             self.mlflow_env.update({"MLFLOW_TRACKING_SERVER_CERT_PATH": client._session.verify})
 
+        mf_full_id = managed_folder
+        if isinstance(managed_folder, DSSManagedFolder):
+            mf_full_id = managed_folder.project.project_key + "." + managed_folder.id
+        elif hasattr(managed_folder, 'name'): # True if dataiku.Folder
+            mf_full_id = managed_folder.name
+
+        if not isinstance(mf_full_id, str):
+            raise TypeError('Type of managed_folder must be "str", "DSSManagedFolder" or "dataiku.Folder".')
```
Comment (lpenet) [inline dataikuapi/dss_plugin_mlflow/utils.py:77]: I would have gone for:
```
if isinstance(managed_folder, DSSManagedFolder):
    mf_full_id = managed_folder.project.project_key + "." + managed_folder.id
elif isinstance(managed_folder, str):
    mf_full_id = managed_folder.name
else:
    try:
        from dataiku import Folder
        if isinstance(managed_folder, Folder):
            mf_full_id = managed_folder.name
    except ImportError:
        pass

if not mf_full_id:
   raise TypeError('Type of managed_folder must be "str", "DSSManagedFolder" or "dataiku.Folder".')
```

**Review by lpenet — CHANGES_REQUESTED** [review]
I would go for a local import and a stricter type checking to also support dataiku.Folder as an authorized input.

File: dataikuapi/dss_plugin_mlflow/utils.py (hunk 1 above)
Comment (alarive) [reply dataikuapi/dss_plugin_mlflow/utils.py]: done and retested by running locally my tests in integration/python/scenario/ml/prediction/mlflow/experiment_tracking/test_client_plugin.py

**Review by lpenet — APPROVED** [review]
LGTM

**Pull request merged**